### ISSUE TITLE
Resource API feedback

### ISSUE THREAD

**Issue by roll** [frictionlessdata/datapackage-php#19]
# Overview

Based on this readme listing I'm adding feedback based on existent implementations and expected lib user competencies (as we target many almost non-tech users - publisher, data wranglers etc).

```php
use frictionlessdata\datapackage;

// add a resource - will raise exception in case of validation error for the new descriptor
$resource = Factory::resource((object)[
    "name" => "new-resource", "data" => ["tests/fixtures/foo.txt", "tests/fixtures/baz.txt"]
])
$datapackage->addResource($resource);
```

### General

See - https://github.com/frictionlessdata/datapackage-php/issues/18

### Resource properties

Implementations reference provides some useful properties (https://github.com/frictionlessdata/implementations):
- `resource.tabular` -> `true` if resource is tabular
- `resource.table` -> to access `tableschema.Table` for tabular resource
- `source/source_type` (this two are still under consideration)

**Comment by OriHoch** [comment]
getting the underlying table object is possible, for each data stream (each data stream object corresponds to a path or data element)

```php
// get the tableschema table object for the first path
$table = $resource->dataStreams()[0]->table
```

I'll add a comment to #22 to add this detail to the README

**Comment by OriHoch** [comment]
fixed the general comments in [v0.1.4](https://github.com/frictionlessdata/datapackage-php/releases)
see the [README](https://github.com/frictionlessdata/datapackage-php/blob/master/README.md) for the updated usage examples

opened an issue to add the additional properties / methods to Resource object: #23
